### PR TITLE
rollbackInvalid & rollbackProperty functions

### LINKED ISSUE #142: Support `rollback` for individual properties
Related to #48.

### PR DESCRIPTION
**Pull request by XaserAcheron**
Though I'm pretty over the moon about this addon's approach to validations, I ran into a case where I needed to ditch all invalid changes and keep (+execute) just the valid ones, but couldn't find a good way to do it with the provided API.

To that extent, here's two new functions:
- `rollbackInvalid()`, which rolls back _just_ the changes that fail validation, and
- `rollbackProperty(key)`, which rolls back a single property by name (closes #142).

Tests and docs included.

I was a bit unsure whether to make `rollbackProperty` a new function or just extend `rollback` to optionally accept a key parameter (similar to `validate`) -- I went the new function route, but will update it to go the other way if desired.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -426,14 +426,41 @@ Note that both changesets `A` and `B` are not destroyed by the merge, so you mig
...
-Rollsback all unsaved changes and resets all errors.
+Rolls back all unsaved changes and resets all errors.
 
 ```js
 changeset.rollback(); // returns changeset
 ```
 
 **[⬆️ back to top](#api)**
 
+#### `rollbackInvalid`
```
Comment (poteto) [inline README.md]: Could you also add an entry in the `API` TOC above?

**Comment by XaserAcheron** [comment]
Oops! Missed a spot -- fixed now. :P

Thanks for taking a look!

**Comment by urbany** [comment]
Thanks @XaserAcheron was just looking for a way to rollback a single attribute.

**Comment by XaserAcheron** [comment]
Glad to help!

At risk of slight annoyance, pinging @poteto for re-review. Should all be kosher now.

**Comment by XaserAcheron** [comment]
Just checking back in on this -- dropping off my monthly @poteto ping, I suppose. :P

**Comment by Dhaulagiri** [comment]
Anything we can do to get this landed? `rollbackProperty` would be really helpful

**Comment by XaserAcheron** [comment]
It's really unfortunate this has been delayed for nearly a year because of a single-line documentation issue that was corrected within a day. 9_6

**Comment by nucleartide** [comment]
Will try to take a look at this over the weekend (but no promises since other things might come up). Sorry it's taken so long!

**Comment by Dhaulagiri** [comment]
@XaserAcheron would you mind rebasing this so it's ready to go?

**Comment by XaserAcheron** [comment]
Merged n' pushed. Tests worked on my end -- we'll see if Travis behaves today. :P

**Comment by XaserAcheron** [comment]
Woohoo! Thanks there.

**Pull request merged**